src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -148,9 +148,19 @@ trait Erasure {
         apply(atp)
       case ClassInfoType(parents, decls, clazz) =>
         ClassInfoType(
-          if (clazz == ObjectClass || isPrimitiveValueClass(clazz)) Nil
+          if (clazz == ObjectClass || isPrimitiveValueClass(clazz) || parents.isEmpty) Nil
```

Comment:
This would look a little clearer / more efficient to me.

```scala
      case ClassInfoType(parents, decls, clazz) if !clazz.isPackageClass =>
        val newParents = if (clazz == ObjectClass || isPrimitiveValueClass(clazz)) Nil
          else if (clazz == ArrayClass) ObjectTpe :: Nil
          else {
            val erasedParents = parents mapConserve this

            // drop first parent for traits -- it has been normalized to a class by now,
            // but we should drop that in bytecode
            if (clazz.hasFlag(Flags.TRAIT) && !clazz.hasFlag(Flags.JAVA))
              ObjectTpe :: erasedParents.tail.filter(_.typeSymbol != ObjectClass)
            else erasedParents
          }
        if (newParents eq parents) tp
        else ClassInfoType(newParents, decls, clazz)
```

Which avoids a needless additional `ClassInfoType` instance when erasure of parents is the identity. That isn't a particular common case as Scala defined classes always erase `AnyRef` to `Object`.

But for Java defined monomorphic hierarchies, it does save the copy.

```
scala> exitingErasure(symbolOf[NoSuchMethodError].info.parents) eq enteringErasure(symbolOf[NoSuchMethodError].info.parents)
res16: Boolean = true
```

Anyway, this problem predated your changes, so feel free to save this suggestion for another time.